### DISCUSSION TITLE
Feature suggestion: filter snapshots in Repository => Tree View

### DISCUSSION THREAD

**Discussion by ovizii** [garethgeorge/backrest#124] [Ideas]
I think this is necessary as one can have one repo and multiple machines backing up different data with different paths into it. Sometimes a snapshot backs up a single path, sometimes multiple ones. At the current state of backrest this makes it pretty difficult say if I wanted to restore a certain path. How could I find it except checking each and every snapshot manually.

Also, could you add the option to backrest to add host, username, and tags? I guess restic ads the first two automatically?

Backrest => Repository => Tree View 

Filter by host, username or tags

![image](https://github.com/garethgeorge/backrest/assets/837902/ceb818fc-2e62-42f3-be52-aef19553c19d)

**EDIT:**
I just realized, the same filter could be pretty useful in the "Operations List" tab.

**Comment by garethgeorge** [comment]
At the moment the primary grouping that backrest provides is by plan. You can get backrest to group backups created by external scripts if you add a tag formatted as `plan:<some unique ID>`. Those backups will then be grouped by the plan IDs you attach.

Backrest aims to be a little opinionated w.r.t. how backups should be structured, so if backing up different paths you should consistently use a different `plan:<some ID>` tag to differentiate those backup sets to make restores easy.

On the filtering front, I can see supporting arbitrary tag filtering but to keep the logic simple I think I'd _only_ want to support tag filtering, likely not other properties of the backup. At the moment this feels rather low priority b/c there's a work around using the `plan:` tag prefix for grouping but I'm interested to get a sense of how much demand for this there is.

**Comment by ovizii** [comment]
Hm, thanks for explaining. Let me wrap my head around this and do some experiments before I get back with feedback or most probably more questions ;-)
